### DISCUSSION TITLE
What is the policy for message discarding?

### DISCUSSION THREAD

**Discussion by maoweifan** [emqx/emqx#10184] [Q&A] [answered]
What is the policy for message discarding?
How many seconds are not used by subscribers before being discarded?
Is this configurable?

**Comment by thalesmg** [comment, marked as answer]
Hi @maoweifan .  The answer depends on which messages we're talking about.

For messages to topics for which there are QoS > 0 subscriptions, then it also depends on whether MQTT v5 is used or not.  This article might be helpful for explaining that: https://www.emqx.com/en/blog/mqtt5-new-feature-clean-start-and-session-expiry-interval

If we're talking about Retained messages, then that's a separate configuration.  See: https://www.emqx.io/docs/en/v5.0/dashboard/retained.html

**Comment by maoweifan** [comment]
hi @thalesmg  First of all thank you very much for helping me to solve my problem

**Comment by maoweifan** [comment]
<img width="410" alt="image" src="https://user-images.githubusercontent.com/51221758/226630230-59e9d0f6-45b7-4587-a2a2-961e326e4d4b.png">

**Comment by maoweifan** [comment]
The version I am using is 5.0.8
My current requirement is that I want to control the message discard
For example, if I set a certain configuration in the configuration file (if any), such as setting 20 seconds, then it will be discarded according to 20 seconds
May I ask if my requirement can be realized?

**Reply by thalesmg** [reply]
If your clients are using MQTT v5 as described in the blog post, you can set the expiry interval using the [`mqtt.session_expiry_interval` config](https://www.emqx.io/docs/en/v5.0/configuration/configuration-manual.html#mqtt-basic-parameters).

**Comment by maoweifan** [comment]
The second point, what I want to know is, can this message be notified to my server program when it is discarded?

**Reply by thalesmg** [reply]
Currently, there's no support for notifications for expired sessions/messages.

**Comment by maoweifan** [comment]
Thank you again for your answer, I look forward to solving my problem, thank you
